File: backend/importar_excel.js
// Script para importar datos de un archivo Excel (.xlsx) a la API usando solicitudes HTTP
// Ejecutar dentro del contenedor backend o con Node.js local
const axios = require('axios');
const XLSX = require('xlsx');
const path = require('path');

const API_URL = process.env.API_URL || 'http://node-backend:8000'; // Cambia si usas otro host/puerto
const EXCEL_FILE = path.join(__dirname, 'ejemplo_importacion.xlsx');

async function importarSheet(sheetName, endpoint) {
  const workbook = XLSX.readFile(EXCEL_FILE);
  const sheet = workbook.Sheets[sheetName];
  if (!sheet) {
    console.error(`No se encontró la hoja: ${sheetName}`);
    return;
  }
  const data = XLSX.utils.sheet_to_json(sheet);
  for (const row of data) {
    try {
      await axios.post(`${API_URL}${endpoint}`, row);
      console.log(`Importado en ${endpoint}:`, row);
    } catch (e) {
      console.error(`Error importando en ${endpoint}:`, row, e.response?.data || e.message);
    }
  }
}

(async () => {
  await importarSheet('signature', '/signature/create');
  await importarSheet('Program', '/program/create');
  console.log('Importación finalizada.');
})();

File: backend/importar_excel.py
import requests
import openpyxl
import os

API_URL = os.environ.get('API_URL', 'http://localhost:8000')
EXCEL_FILE = os.path.join(os.path.dirname(__file__), 'ejemplo_importacion.xlsx')

# Cargar el archivo Excel
wb = openpyxl.load_workbook(EXCEL_FILE)

# Importar signature
if 'signature' in wb.sheetnames:
    ws = wb['signature']
    headers = [cell.value for cell in ws[1]]
    for row in ws.iter_rows(min_row=2, values_only=True):
        data = dict(zip(headers, row))
        try:
            r = requests.post(f'{API_URL}/signature/create', json=data)
            print('Signature:', data, '->', r.status_code, r.text)
        except Exception as e:
            print('Error importando signature:', data, e)
else:
    print('No se encontró la hoja signature')

# Importar Program
if 'Program' in wb.sheetnames:
    ws = wb['Program']
    headers = [cell.value for cell in ws[1]]
    for row in ws.iter_rows(min_row=2, values_only=True):
        data = dict(zip(headers, row))
        try:
            r = requests.post(f'{API_URL}/program/create', json=data)
            print('Program:', data, '->', r.status_code, r.text)
        except Exception as e:
            print('Error importando program:', data, e)
else:
    print('No se encontró la hoja Program')
